### PR TITLE
Add setting Windows resolution step for prereleases testing workflow and some tests fixes

### LINKED ISSUE #7357: [test-bot] pip install --pre is failing
The --pre Test workflow failed on 2024-11-13 12:24 UTC

The most recent failing test was on windows-latest py3.12 pyqt6
with commit: d05d688c266fdcdd1d8ee437e8eb6cad8ef71fa7

Full run: https://github.com/napari/napari/actions/runs/11816924339

(This post will be updated if another test fails, as long as this issue remains open.)

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

Closes #7357

# Description

* Missing workflow step on the prereleases testing workflow to prevent generating warnings when testing fullscreen toggle on Windows
* Use `qtbot.waitUntil` and `qtbot.wait` for delay of macOS animation when changing from/to fullscreen window state over the maximized window state test
* Don't check for menu visibility if native menu bar is being used (macOS for example)

### CODE REVIEW

File: napari/_qt/qt_main_window.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ def eventFilter(self, source, event):
                         self.menuBar().hide()
             elif event.type() == QEvent.Type.Leave and source is self:
                 self.menuBar().hide()
-        return super().eventFilter(source, event)
+        return False
```
Comment (Czaki) [inline napari/_qt/qt_main_window.py]: why this change?

Comment (dalthviz) [reply napari/_qt/qt_main_window.py]: I have been seeing around a test failing with PySide6 due to the handling of a `QWidgetItem` event when reaching this line. Errors like https://github.com/napari/napari/actions/runs/11783830702/job/32824335938#step:15:249 and https://github.com/napari/napari/actions/runs/11784483916/job/32824348990?pr=7348#step:15:249

So I was checking if just not calling `super` here could make things pass. However, maybe checking what is going on with that over a separate PR is better?

Comment (dalthviz) [reply napari/_qt/qt_main_window.py]: Just in case, with the last commit reverted this change and to explore the error mentioned I opened PR #7367

Edit: PR #7368 takes care of the tests failing on PySide6 by skipping the Linux + Python 3.9 + PySide6 job combination

**Review by Czaki — APPROVED** [review]
